test/create.js

```diff
@@ -76,6 +77,36 @@ test('custom headers (extend)', async t => {
 	t.is(headers.unicorn, 'rainbow');
 });
 
+test('extend overwrites arrays', t => {
+	const statusCodes = [408];
+	const a = got.extend({retry: {statusCodes}});
+	t.deepEqual(a.defaults.options.retry.statusCodes, statusCodes);
+	t.is(a.defaults.options.retry.statusCodes, statusCodes);
+});
+
+test('extend removes object values set to undefined', t => {
+	const a = got.extend({
+		headers: {foo: 'foo', bar: 'bar', baz: 'baz'}
+	});
+	const b = a.extend({headers: {foo: undefined, bar: null}});
+	t.deepEqual(
+		b.defaults.options.headers,
+		{...got.defaults.options.headers, ...{baz: 'baz', bar: null}}
+	);
+});
```

Comment:
:-1: The feature is meant **only** for `headers`. Otherwise, if the new value is `undefined`, it shouldn't be replaced.

Reply:
Ack. This is the same behavior as we're discussing above, so we'll resolve this when we resolve that.